File: app/src/main/java/cz/feldis/gasprices/models/GasPricesResponse.kt
package cz.feldis.gasprices.models

import com.google.gson.annotations.SerializedName

data class GasPricesResponse(
    val version: String,
    @SerializedName("class") val categoryClass: String,
    val label: String,
    val update: String,
    val href: String,
    val dimension: Dimension,
    val value: List<Float>
)

data class Dimension(
    val sp0207ts_tyz: Category,
    val sp0207ts_ukaz: Category,
    val sp0207ts_data: Category
)

data class Category(
    val label: String,
    val note: String,
    val category: CategoryDetail
)

data class CategoryDetail(
    val index: Map<String, Int>,
    val label: Map<String, String>
)

File: ios/GasPricesiOS/Data/Models/GasPricesResponse.swift
import Foundation

struct GasPricesResponse: Decodable {
    let version: String
    let categoryClass: String
    let label: String
    let update: String
    let href: String
    let dimension: Dimension
    let value: [Double?]

    enum CodingKeys: String, CodingKey {
        case version
        case categoryClass = "class"
        case label
        case update
        case href
        case dimension
        case value
    }
}

struct Dimension: Decodable {
    let sp0207ts_tyz: Category
    let sp0207ts_ukaz: Category
    let sp0207ts_data: Category
}

struct Category: Decodable {
    let label: String
    let note: String?
    let category: CategoryDetail

    func orderedLabels() -> [String] {
        category.orderedLabels()
    }
}

struct CategoryDetail: Decodable {
    let index: [String: Int]
    let label: [String: String]

    func orderedLabels() -> [String] {
        if !index.isEmpty {
            return index
                .sorted { $0.value < $1.value }
                .compactMap { label[$0.key] }
        }

        return label
            .sorted { $0.key < $1.key }
            .map(\.value)
    }
}